tests/apps/service_invocation_external/app.go

```diff
@@ -103,6 +108,18 @@ func init() {
+		// #nosec
+		os.WriteFile("/tmp/testdata/certs/key.pem", []byte("-----BEGIN PRIVATE KEY-----\nMIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCdsj8z9eHsZWgm\n95vwpWLUNZxVEO3ME4bzAltYksSZUuJOJz9/xroFFvbuUE8ToRFBloajD0xR7pJw\nmxuTWIYFKG+/2tIbZwSzfOm/7UHRa5LBlbtn6skWaoixbOJcOX0r1s0GE2R0yRWo\nDyo2QTI/S2FJqcDFXMMNErAKGfqaez9+K/Bq9XHhmKAZClFi142KvBsaliyZ8L/Z\npgOz8F1XwcGgUwTRRrNeNapKYNzqy9FgSwtyAy5Mc/GfHmaIap5fFWgsVs/ywmZS\nqLSc9x6KzhZPQ6SR4lR3j2+AN+4q7l0XuVy03/kkFDMkhf/vjS167HdHwuXkUs/v\neD3m22pXAgMBAAECggEADiTMkmZ9SuyVeNlAfUb4JDxityZ7jZA6zFTryHfxdib+\n8S+H/Hfge1Bh+WHulb4ehBlyHfHjUH3rWHow8PhBnPjRfQ2tRpPo0ghkFri67IOP\neBlr45glTQNlGzS+4LKvTV1IqkDXA7p9XMNvb8pNLWwfX7H//jOhsux4KY/G8E/1\niQUONGIfbYU2o4ConRKfVd8WklDWx0hlKID4GgNaHcaMFyJwFai6TmPIezX0YVV/\n3KkS5U9Y7IH/Vh5lnzsl8KAwpum7nU04qsCMb9UC/BAEukEdYilk88z3axadhMxu\n6shAqUUG+MiwZRi3Z8mdIGXFin7iZJAb25cFDEMq4QKBgQDJ1H/xgnGV/8gakNmX\nAcuczu33qqLNBnwpFLHaTHlXFvTJ3rkXHfkgai/j+XDPI5zWD+2I23OmwKfwL9tl\naMO7823aP39UJxjEtM6kWiO6TBpx2xsq7LdvjZmpxIlBBRxlNwd0vXlWDaOIC8Bq\nuI/nZH3F5dEj/yneZPvYiNS6bwKBgQDIBV374lD/hfp/wH2MkudWth//UJ5mGoDZ\n+Y8uadcuntg1ydpreZrABk+Do1HNhqT0xi1GCKgre6R3iw9Gj+Vf3FytD34XZ/qN\nfyNai2hkSQYCQNxGSPonfyxrwHi7dVlA/hXW0+A5cCwnJJHSKn0wvx1XCyt4Df11\nx1Vut/7imQKBgBIhO4j3PbrKqi/wuQr+h0rDr86ijlM6dOa/zCYe3nt7PWvwzNGN\nyLYns7uXyZFh0pHMBntXszDUd4F02xFl8scsW77SG1700B3DXBiZXEI82ddGZk9Z\nyjSEFLySQf4MNnvFEK9k4cRgH0N2QseTYqHXfhDLzxibsCghDBpsWCeTAoGAU9n8\nJmYf8p7l/rxhZZcIKE+9ZjkTzlX6AmceYSS5sq+gkBwI1tlpyGpc1yCWYIS7jmmt\n/RgP9/qjXgIL61bTn7Yvm9VM1N8gKEVOOt0KbFA4lqU+MrDRPdGQe7N6aYuuWRKr\ncR3td+bQe7BW+TYauV7FHAJwvKmQwdlItVupHLECgYAw7vY/px+nY0gFwjdPhfnz\ncQu+NOqCs5l7ibY5f53D76rZJDkUGm4fT3bejuL64eeNNI3EHRdcQ5ayeYEbuzy+\nwbiAgaYIlsNH0aVIWkzv1hCbOtXEl7o7uGv7FWVjEvhen865dGCeUD9kGpeePEfM\nJgvfq8dv6ZULV+5OFVpOPg==\n-----END PRIVATE KEY-----\n"), 0o600)
```

Comment:
Is there a very strong reason to commit a private key here? Key generation is very cheap, especially using types like ecdsa P256. Committing private keys is dangerous, particularly in a post LLM worlds, and leads to situations like [this](https://programmingwithstyle.com/posts/howihackedmycar/).

Reply:
What's an alternative? The same key needs to be used in other places for the test and can't be generated on the fly

Reply:
Writing private key and certificate material to a Kubernetes Secret which is mounted by the application.

Reply:
Done